lib/integrations/awsoidc/aws_app_access_iam_config.go

```diff
@@ -0,0 +1,127 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package awsoidc
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/gravitational/trace"
+
+	awslib "github.com/gravitational/teleport/lib/cloud/aws"
+)
+
+const (
+	// defaultPolicyNameForAWSAppAccess is the default name for the Inline Policy added to the IntegrationRole.
+	defaultPolicyNameForAWSAppAccess = "AWSAppAccess"
+)
+
+// AWSAppAccessConfigureRequest is a request to configure the required Policies to use AWS App Access.
+type AWSAppAccessConfigureRequest struct {
+	// IntegrationRole is the Integration's AWS Role used to set up Teleport as an OIDC IdP.
+	IntegrationRole string
+
+	// IntegrationRoleAWSAppAccessPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to AWSAppAccess
+	IntegrationRoleAWSAppAccessPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *AWSAppAccessConfigureRequest) CheckAndSetDefaults() error {
+	if r == nil {
+		return trace.BadParameter("request is nil")
+	}
+
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.IntegrationRoleAWSAppAccessPolicy == "" {
+		r.IntegrationRoleAWSAppAccessPolicy = defaultPolicyNameForAWSAppAccess
+	}
+
+	return nil
+}
+
+// AWSAppAccessConfigureClient describes the required methods to create the IAM Policies required for AWS App Access.
+type AWSAppAccessConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+type defaultAWSAppAccessConfigureClient struct {
+	*iam.Client
+}
+
+// NewAWSAppAccessConfigureClient creates a new AWSAppAccessConfigureClient.
+func NewAWSAppAccessConfigureClient(ctx context.Context) (AWSAppAccessConfigureClient, error) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if cfg.Region == "" {
+		// IAM Service does not support regions, however a value is required:
+		// https://github.com/aws/aws-sdk-go-v2/issues/1778#issuecomment-1210031692
+		// Providing an invalid region here, ensures the service uses the default AWS Partition.
+		cfg.Region = " "
+	}
+
+	return &defaultAWSAppAccessConfigureClient{
+		Client: iam.NewFromConfig(cfg),
+	}, nil
```

Comment:
Could you do this:

```suggestion
	return iam.NewFromConfig(cfg), nil
```

and drop the defaultAWSAppAccessConfigureClient type?

Reply:
Yes :+1: